imap_processing/swe/tests/test_swe_l1b.py

```diff
@@ -0,0 +1,167 @@
+    # into order.
+    sorted_packets = sorted(
+        grouped_data[1344], key=lambda x: x.data["QUARTER_CYCLE"].raw_value
+    )
+    science_l1a_ds = swe_science(sorted_packets)
+    # convert value from raw to engineering units as needed
+    conversion_table_path = (
+        f"{imap_module_directory}/swe/l1b/engineering_unit_convert_table.csv"
+    )
+    # Look up packet name from APID
+    packet_name = SWEAPID.SWE_SCIENCE.name
+    # Convert raw data to engineering units as needed
+    science_l1b = convert_raw_to_eu(
+        science_l1a_ds,
+        conversion_table_path=conversion_table_path,
+        packet_name=packet_name,
+    )
+
+    # read science validation data
+    test_data_path = f"{imap_module_directory}/swe/tests/l0_validation_data"
+    eu_validation_data = pd.read_csv(
+        f"{test_data_path}/idle_export_eu.SWE_SCIENCE_20230927_172708.csv",
+        index_col="SHCOARSE",
+    )
+    second_data = sorted_packets[1]
+    validation_data = eu_validation_data.loc[second_data.data["SHCOARSE"].raw_value]
+
+    science_eu_field_list = [
+        "SPIN_PHASE",
+        "SPIN_PERIOD",
+        "THRESHOLD_DAC",
+    ]
+
+    # Test EU values for science data
+    for field in science_eu_field_list:
+        assert round(science_l1b[field].data[1], 5) == round(validation_data[field], 5)
+
+    # process housekeeping data to l1a and create l1b
+    hk_l1a_ds = create_dataset(grouped_data[1330])
+    hk_l1b = convert_raw_to_eu(
+        hk_l1a_ds, conversion_table_path, SWEAPID.SWE_APP_HK.name
+    )
+
+    # read housekeeping validation data
+    eu_validation_data = pd.read_csv(
+        f"{test_data_path}/data_derived.SWE_APP_HK_20230927_094839.csv",
+        index_col="SHCOARSE",
+    )
+    second_data = grouped_data[1330][1]
+    validation_data = eu_validation_data.loc[second_data.data["SHCOARSE"].raw_value]
+
+    # check that these field's calculated EU value matches with
+    # validation data's EU value.
+    eu_field_list = [
+        "HVPS_VBULK",
+        "HVPS_VCEM",
+        "HVPS_VESA",
+        "HVPS_VESA_LOW_RANGE",
+        "HVPS_ICEM",
+        "FEE_TEMP",
+        "SENSOR_TEMP",
+        "HVPS_TEMP",
+        "LVPS_BACK_BOARD_TEMP",
+    ]
+
+    # Check EU values from housekeeping data
+    for field in eu_field_list:
+        assert round(hk_l1b[field].data[1], 5) == round(validation_data[field], 5)
+
+
+def test_cdf_creation(l1a_test_data, cdf_data):
+    sci_l1b_filepath = swe_l1b(l1a_test_data)
+
+    # process hk data to l1b
+    hk_data = [
+        ds for ds in cdf_data if ds["PKT_APID"].data[0] == SWEAPID.SWE_APP_HK.value
+    ]
+    hk_l1b_filepath = swe_l1b(hk_data[0])
+
+    # process evtmsg data to l1b
+    evtmsg_data = [
+        ds for ds in cdf_data if ds["PKT_APID"].data[0] == SWEAPID.SWE_EVTMSG.value
+    ]
+    evtmsg_l1b_filepath = swe_l1b(evtmsg_data[0])
+
+    assert os.path.basename(sci_l1b_filepath) == "imap_swe_l1b_20230927_sci_v01.cdf"
```

Comment:
Are these `Path` objects? if so, you can just get the name. If not, maybe you can make the return be a path object instead of a string path?
```suggestion
    assert sci_l1b_filepath.name == "imap_swe_l1b_20230927_sci_v01.cdf"
```

Reply:
It looks like it's not `Path` object but a string.